avr/cores/megacommand/MCL/ExtSeqTrack.cpp

```diff
@@ -31,6 +29,69 @@ void ExtSeqTrack::re_sync() {
   mute_until_start = true;
 }
 
+void ExtSeqTrack::remove_event(uint16_t index) {
+  uint16_t ev_idx = 0;
+  uint8_t step;
+  uint8_t bucket = 0;
+  for (step = 0; step < length && ev_idx <= index; ++step) {
+    bucket = timing_buckets.get(step);
+    ev_idx += bucket;
+  }
+  // step is at least 1, move back to get current step
+  --step;
+  timing_buckets.set(step, bucket - 1);
+  memmove(events + index, events + index + 1,
+          sizeof(ext_event_t) * (NUM_EXT_EVENTS - index - 1));
+  --event_count;
+}
+
+uint16_t ExtSeqTrack::add_event(uint8_t step) {
+  uint8_t u = timing_buckets.get(step);
```

Comment:
shouldn't this line, be placed after the memmove.

The locate(step,idx,end) command will return u + 1